### PR TITLE
redesign the community card in home page

### PR DESCRIPTION
**Pull request by KaleemNeslit**
## Link to Issue
Closes: #9984 

## Description of Changes
- Update the Trending Communities card in Homepage 
- now user can join the community from the card 

## "How We Fixed It"

- Update the Trending Communities card in Homepage 
- now user can join the community from the card 

## Test Plan
In home screen look for Trending Communities in right side you will be able to join community from there if not already joined

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/CommunityPreviewCard/CommunityPreviewCard.tsx [hunk 1]
```diff
@@ -6,15 +11,30 @@ import { CWCard } from '../../../../components/component_kit/cw_card';
...
+    base: ChainBase;
+  };
   monthlyThreadCount?: number;
   isCommunityMember?: boolean;
   hasNewContent?: boolean;
   onClick?: () => any;
   isExploreMode?: boolean;
-};
+} & (
+  | { isExploreMode: true }
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/CommunityPreviewCard/CommunityPreviewCard.tsx:27]: plz move `community?:` from above here

**Comment by mzparacha** [comment]
1. For communities with larger names, i get some overflows
image

2. If a click on "Join community" button when non-auth state, can we make it "User click -> auth modal opens -> user logs in -> user joins community"? There is a `useDeferredConditionTriggerCallback` hook that is specifically built for this purpose.

https://github.com/user-attachments/assets/9a776e40-221b-4b16-8a43-09ebc5c1ed10

3. lets add some pad b/w the thread/m and "new" tag
image

**Comment by masvelio** [comment]
1. Please always check your PRS how it is looking on mobile. Our common approach is truncating names to 1 or max 2 lines. Use `multiline-text-ellipsis`
image

**Comment by Israellund** [comment]
@KaleemNeslit You could maybe use `smartTrim()` to dictate the length of the community names so they don't mess with the design.

**Comment by KaleemNeslit** [comment]
@masvelio @mzparacha  i made the ui changes  kindly review it .
there is some other issues like useDeferredConditionTriggerCallback is not working as expected i will create the new ticket for that

**Review by masvelio — APPROVED** [review]

**Pull request merged**